internal/docstore/dynamodocstore/dynamo.go

```diff
@@ -202,13 +202,27 @@ func newCollection(db *dyn.DynamoDB, tableName, partitionKey, sortKey string, op
 	}, nil
 }
 
-func (c *collection) KeyFields() []string {
-	if c.sortKey == "" {
-		return []string{c.partitionKey}
+// Key returns a two-element array with the partition key and sort key, if any.
+func (c *collection) Key(doc driver.Document) (interface{}, error) {
+	var keys [2]interface{}
+	var err error
+	keys[0], err = doc.GetField(c.partitionKey)
+	if err != nil {
+		return nil, nil // missing key is not an error
 	}
-	return []string{c.partitionKey, c.sortKey}
+	if c.sortKey != "" {
+		keys[1], _ = doc.GetField(c.sortKey) // ignore error since keys[1] is nil in that case
+	}
+	return keys, nil
 }
 
+// func (c *collection) KeyFields() []string {
```

Comment:
What is this method? Should it be removed?

Reply:
Done.